src/vs/workbench/contrib/terminal/browser/terminalActions.ts

```diff
@@ -1402,13 +1402,18 @@ export function registerTerminalActions() {
 		}
 		async run(accessor: ServicesAccessor, profile?: ITerminalProfile) {
 			const terminalService = accessor.get(ITerminalService);
+			const commandService = accessor.get(ICommandService);
 			await terminalService.doWithActiveInstance(async t => {
 				const cwd = await getCwdForSplit(terminalService.configHelper, t, accessor.get(IWorkspaceContextService).getWorkspace().folders, accessor.get(ICommandService));
 				if (cwd === undefined) {
 					return undefined;
 				}
-				terminalService.splitInstance(t, profile, cwd);
-				return terminalService.showPanel(true);
+				if (t.target !== TerminalLocation.Editor) {
```

Comment:
nit: Since we're covering both it would make the code simpler to make the if `===`